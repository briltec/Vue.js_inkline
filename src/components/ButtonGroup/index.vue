<template src="./template.html" />
<script src="./script.js" />
<style src="../Button/style.styl" lang="stylus" scoped />
<style src="./style.styl" lang="stylus" scoped />
