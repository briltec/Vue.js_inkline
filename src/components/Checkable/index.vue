<script src="./script.js" />
<!--<style src="./style.styl" lang="stylus" />-->
