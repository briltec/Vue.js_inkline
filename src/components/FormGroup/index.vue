<template src="template.html" />
<script src="script.js" />
<style src="../Input/style.styl" lang="stylus" scoped />
<style src="style.styl" lang="stylus" scoped />
