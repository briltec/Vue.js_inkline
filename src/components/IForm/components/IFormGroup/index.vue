<template src="./template.html" />
<style src="./style.scss" lang="scss" />
<script src="./script.js" />
