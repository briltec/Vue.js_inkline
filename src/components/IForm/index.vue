<template src="./template.html" />
<script src="./script.ts" lang="ts" />
<style src="../IInput/style.scss" lang="scss" scoped />
<style src="./style.scss" lang="scss" scoped />
