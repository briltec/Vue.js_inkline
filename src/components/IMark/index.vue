<template src="./template.html" />
<script src="./script.ts" lang="ts" />
