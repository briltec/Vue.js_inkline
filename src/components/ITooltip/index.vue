<template src="./template.html" />
<script src="./script.ts" lang="ts" />
<style src="./style.scss" lang="scss" />
