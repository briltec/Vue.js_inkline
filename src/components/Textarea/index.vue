<template src="./template.html" />
<style src="../Input/style.styl" lang="stylus" />
<script src="./script.js" />
