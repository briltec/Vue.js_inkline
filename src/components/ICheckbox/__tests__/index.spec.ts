import { shallowMount } from '@vue/test-utils';
import {IButton, ICheckbox} from '@inkline/inkline/components';

describe('Components', () => {
    describe('ICheckbox', () => {
        const props = {
            color: 'light',
            size: 'md'
        };

        it('should be named correctly', () => {
            expect(ICheckbox.name).toEqual('ICheckbox');
        });

        it('should render correctly', () => {
            const wrapper = shallowMount(ICheckbox, { props });
            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('computed', () => {
            describe('classes', () => {
                it('should return classes object', () => {
                    const wrapper: any = shallowMount(ICheckbox, {
                        props: {
                            disabled: true,
                            readonly: true,
                            native: true,
                            ...props
                        }
                    });

                    expect(wrapper.vm.classes).toEqual({
                        [`-${wrapper.vm.color}`]: true,
                        [`-${wrapper.vm.size}`]: true,
                        '-disabled': true,
                        '-readonly': true,
                        '-native': true
                    });
                });
            });

            describe('checked', () => {
                it('should return true if formGroup.checked contains value', () => {
                    const value = 'value';
                    const wrapper: any = shallowMount(ICheckbox, {
                        global: {
                            provide: {
                                formGroup: {
                                    checked: [value]
                                }
                            }
                        },
                        props: {
                            value,
                            ...props
                        }
                    });

                    expect(wrapper.vm.checked).toEqual(true);
                });
            });

            describe('tabIndex', () => {
                it('should return -1 if disabled', () => {
                    const wrapper: any = shallowMount(IButton, {
                        props: {
                            disabled: true,
                            ...props
                        }
                    });

                    expect(wrapper.vm.tabIndex).toEqual(-1);
                });

                it('should return tabindex otherwise', () => {
                    const wrapper: any = shallowMount(IButton, { props });

                    expect(wrapper.vm.tabIndex).toEqual(wrapper.vm.tabindex);
                });
            });
        });
    });
});
