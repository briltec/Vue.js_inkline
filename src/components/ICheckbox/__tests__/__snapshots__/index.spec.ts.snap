// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ICheckbox should render correctly 1`] = `<div class="checkbox -light -md" role="checkbox"><input type="checkbox" tabindex="-1" name="checkbox"><label class="checkbox-label" aria-checked="false" tabindex="1"></label></div>`;
